[Bug] EJB Container crashes during deployment when using @ActivationConfigProperty for resourceAdapter

### GlassFish Version (and build number)

7.0.23 (Docker Image: ghcr.io/eclipse-ee4j/glassfish:7.0.23)

### JDK version

JDK 17

### OS

Docker Desktop (Windows host)

### Database

MySQL 8.0.33

### Problem Description

**Current Behavior:**
When attempting to deploy a Jakarta EE application containing a Message-Driven Bean (MDB) to the `ghcr.io/eclipse-ee4j/glassfish:7.0.23` Docker image, the EJB container initialization fails entirely if the `@ActivationConfigProperty` for `resourceAdapter` is specified in the Java code. 

The deployment is aborted, throwing an `IllegalArgumentException` stating that the type for the environment property is illegal.

**Expected Behavior:**
The MDB should successfully parse the string `"jmsra"`, bind to the internal JMS resource adapter, and allow the application to deploy, as it did in older `javax` versions of GlassFish.

**Stack Trace Snippet:**
```text
[#|2026-05-24T18:58:36.844067Z|WARNING|GF 7.0.23|jakarta.enterprise.resource.resourceadapter.com.sun.enterprise.connectors.inbound|_ThreadID=97;_ThreadName=AutoDeployer;_LevelValue=900;_MessageID=endpoint.activation.failure;|
  RAR8501: Exception during endpoint activation for ra [ jmsra ], activationSpecClass [ com.sun.messaging.jms.ra.ActivationSpec ] : java.lang.IllegalArgumentException: Illegal type for environment properties: [interface jakarta.resource.spi.ResourceAdapter]|#]

Caused by: java.lang.IllegalArgumentException: Illegal type for environment properties: [interface jakarta.resource.spi.ResourceAdapter]
        at com.sun.enterprise.deployment.EnvironmentProperty.getObjectFromString(EnvironmentProperty.java:376)
        at com.sun.enterprise.deployment.EnvironmentProperty.getResolvedValueObject(EnvironmentProperty.java:127)
        at com.sun.enterprise.connectors.util.SetMethodAction.run(SetMethodAction.java:77)
```

### Steps to reproduce

1. Create an MDB annotated as follows:

```java
@MessageDriven(
        activationConfig = {
                @ActivationConfigProperty(propertyName = "destinationLookup", propertyValue = "jms/NewUserTopic"),
                @ActivationConfigProperty(propertyName = "destinationType", propertyValue = "jakarta.jms.Topic"),
                @ActivationConfigProperty(propertyName = "resourceAdapter", propertyValue = "jmsra") // <-- Triggers the crash
        }
)
public class MyListener implements MessageListener { 
    @Override
    public void onMessage(Message message) {}
}

```

2. Build the application into a `.war` file.
3. Deploy the application via autodeploy in a fresh GlassFish 7.0.23 Docker container.
4. Observe the `server.log` crash.

**Workaround:**
Removing the `resourceAdapter` property from the Java annotation and instead defining it globally in `glassfish-ejb-jar.xml` allows the application to deploy successfully.

### Impact of Issue

This issue severely impacts the developer experience when migrating legacy applications to Jakarta EE 10 and GlassFish 7. It prevents developers from utilizing standard annotation-based MDB configuration for custom resource adapters, forcing a fallback to XML deployment descriptors. It completely breaks containerized CI/CD pipelines that rely on autodeploy.

--- a/appserver/connectors/connectors-inbound-runtime/src/main/java/com/sun/enterprise/connectors/inbound/ConnectorMessageBeanClient.java
+++ b/appserver/connectors/connectors-inbound-runtime/src/main/java/com/sun/enterprise/connectors/inbound/ConnectorMessageBeanClient.java
@@ -137,7 +137,6 @@ public final class ConnectorMessageBeanClient implements MessageBeanClient, Mess
         ActiveInboundResourceAdapter aira = getActiveResourceAdapter(resourceAdapterMid);
         aira.updateMDBRuntimeInfo(descriptor, this.messageBeanPM.getPoolDescriptor());
 
-        // the resource adapter this MDB client is deployed to
         ResourceAdapter ra = aira.getResourceAdapter();
         if (ra == null) {
             String i18nMsg = localStrings.getString("msg-bean-client.ra.class.not.specified", resourceAdapterMid);
@@ -155,11 +154,16 @@ public final class ConnectorMessageBeanClient implements MessageBeanClient, Mess
             throw new IllegalStateException("Unsupported message listener type");
         }
 
-        if (logger.isLoggable(FINEST)) {
-            logger.log(FINEST, "ActivationSpecClassName = " + activationSpecClassName);
-        }
         try {
-            ActivationSpec activationSpec = getActivationSpec(aira, activationSpecClassName);
+            ActivationSpec activationSpec;
+            try {
+                activationSpec = getActivationSpec(aira, activationSpecClassName);
+            } catch (IllegalArgumentException ex) {
+                String errorMsg = localStrings.getString("msg-bean-client.activation.invalid.property", descriptor.getName());
+                logger.log(Level.SEVERE, errorMsg, ex);
+                throw new ConnectorRuntimeException(errorMsg, ex);
+            }
+
             activationSpec.setResourceAdapter(ra);
 
             // at this stage, activation-spec is created, config properties merged with ejb-descriptor.
@@ -171,9 +175,12 @@ public final class ConnectorMessageBeanClient implements MessageBeanClient, Mess
             this.myState = BLOCKED;
             ra.endpointActivation(this, activationSpec);
             aira.addEndpointFactoryInfo(beanID_, new MessageEndpointFactoryInfo(this, activationSpec));
+
+        } catch (ConnectorRuntimeException cre) {
+            throw cre;
         } catch (Exception ex) {
             logger.log(Level.WARNING, "endpoint.activation.failure",
-                new Object[] {resourceAdapterMid, activationSpecClassName, ex});
+            new Object[] {resourceAdapterMid, activationSpecClassName, ex});
             throw new ConnectorRuntimeException("Setup failed.", ex);
         }
     }

--- a/appserver/connectors/connectors-inbound-runtime/src/main/resources/com/sun/enterprise/connectors/inbound/LocalStrings.properties
+++ b/appserver/connectors/connectors-inbound-runtime/src/main/resources/com/sun/enterprise/connectors/inbound/LocalStrings.properties
@@ -19,3 +19,4 @@ msg-bean-client.multiple-ras-supporting-message-listener=The 'resource-adapter-m
 msg-bean-client.defaulting.message-listener.supporting.rar=No 'resource-adapter-mid' specified, defaulting to the resource-adapter [ {0} ] that supports the requested 'message-listener' [ {1} ]
 msg-bean-client.ra.class.not.specified=<resourceadapter-class> must be specified for the resource-adapter [ {0} ] having inbound (<inbound-resourceadapter> or its equivalent annotation) artifacts
 msg-bean-client.could-not-derive-ra-mid=Unable to determine resourceAdapterMid for descriptor [ {0} ]
+msg-bean-client.activation.invalid.property=Deployment failed for MDB [ {0} ]: Invalid type provided for ActivationConfigProperty.

